### ISSUE TITLE
osm2pgsql syntax warnings

### ISSUE THREAD

**Issue by pnorman** [osm2pgsql-dev/osm2pgsql-themepark#25]
Currently when using a style written for themepark you get this message on recent osm2pgsql versions
```
Processing: Node(3430k 1715.0k/s) Way(0k 0.00k/s) Relation(0 0.0/s)
2024-12-19 13:05:44  WARNING: You should use the syntax 'object:as_polygon()' (with the colon, not a point) to call functions on the OSM object.
Processing: Node(3926k 1963.0k/s) Way(536k 26.80k/s) Relation(0 0.0/s)
2024-12-19 13:06:05  WARNING: You should use the syntax 'object:as_multipolygon()' (with the colon, not a point) to call functions on the OSM obj
ect.
```

This happens regardless of if the old syntax is used in the style

**Comment by joto** [comment]
If you'd post a reproducible test case, I could actually try to debug that problem...

**Comment by lonvia** [comment]
If this is with spirit, then the issue likely comes from there. There are a couple of bad uses of dot. Examples: 

https://github.com/pnorman/spirit/blob/76988dad08b644c24f7fdec45b191e13f782d716/themes/shortbread/topics/water.lua#L160

https://github.com/pnorman/spirit/blob/76988dad08b644c24f7fdec45b191e13f782d716/themes/spirit/topics/buildings.lua#L28

.... and a dozen or so others.

**Comment by pnorman** [comment]
> If you'd post a reproducible test case, I could actually try to debug that problem...

Sorry - I've reduced it now. It was a call of `object.as_area()` in my code which I hadn't noticed. I had verified that there were no uses of `object:as_polygon()` or `object:as_multipolygon()`, but https://github.com/osm2pgsql-dev/osm2pgsql-themepark/blob/master/lua/themepark.lua#L515 renames the function in themepark and the osm2pgsql message was about a different function name.
